python/runtime/cudaq/platform/py_alt_launch_kernel.cpp

```diff
@@ -504,9 +516,26 @@ MlirModule synthesizeKernel(const std::string &name, MlirModule module,
   auto context = cloned.getContext();
   registerLLVMDialectTranslation(*context);
 
+  // Get additional debug values
+  auto disableMLIRthreading = getEnvBool("CUDAQ_MLIR_DISABLE_THREADING", false);
+  auto enablePrintMLIREachPass =
+      getEnvBool("CUDAQ_MLIR_PRINT_EACH_PASS", false);
+
   PassManager pm(context);
   pm.addPass(cudaq::opt::createQuakeSynthesizer(name, rawArgs));
   pm.addPass(createCanonicalizerPass());
+
+  // Run state preparation for quantum devices only.
+  // Simulators have direct implementation of state initialization
+  // in their runtime.
+  auto &platform = cudaq::get_platform();
+  if (!platform.is_simulator() || platform.is_emulated()) {
+    pm.addPass(cudaq::opt::createConstPropComplex());
+    pm.addPass(createCSEPass());
```

Comment:
Do we need CSE or will Canonicalizer suffice?

Reply:
will try without CSE to check it out. I think I needed it for some reason...

Reply:
Actually, neither of them is needed here anymore. I previously had to insert a CSE to cleanup my (incorrect) cast replacement that would sometimes create duplicate casts. Removed, thanks!